lib/screens/v2/widget_instance/evergreen_content.ex

```diff
@@ -22,7 +30,28 @@ defmodule Screens.V2.WidgetInstance.EvergreenContent do
 
   def widget_type(_instance), do: :evergreen_content
 
-  def valid_candidate?(_instance), do: true
+  def valid_candidate?(%__MODULE__{schedule: schedule, now: now}) do
+    schedule
+    |> Enum.map(fn x ->
+      case x do
+        %Schedule{start_dt: nil, end_dt: nil} ->
+          true
+
+        %Schedule{start_dt: start_dt, end_dt: nil} ->
+          DateTime.compare(start_dt, now) in [:lt, :eq]
+
+        %Schedule{start_dt: nil, end_dt: end_dt} ->
+          DateTime.compare(end_dt, now) == :gt
+
+        %Schedule{start_dt: start_dt, end_dt: end_dt} ->
+          DateTime.compare(start_dt, now) in [:lt, :eq] and DateTime.compare(end_dt, now) == :gt
+
+        _ ->
+          false
+      end
+    end)
```

Comment:
`fn` allows you to provide multiple clauses in the same way that `def`/`defp` do—you can remove one level of indentation here like so:
```suggestion
    |> Enum.map(fn
      %Schedule{start_dt: nil, end_dt: nil} ->
        true

      %Schedule{start_dt: start_dt, end_dt: nil} ->
        DateTime.compare(start_dt, now) in [:lt, :eq]

      %Schedule{start_dt: nil, end_dt: end_dt} ->
        DateTime.compare(end_dt, now) == :gt

      %Schedule{start_dt: start_dt, end_dt: end_dt} ->
        DateTime.compare(start_dt, now) in [:lt, :eq] and DateTime.compare(end_dt, now) == :gt

      _ ->
        false
    end)
```

Also, it should be safe to remove the `_ ->` fallback clause—if we get a value that isn't at all what we were expecting, we'd rather just let this process crash, rely on automatic Splunk logging to track down where things went wrong, and fix the problem. In general Elixir development encourages a ["let it crash" or "fail fast"](https://elixir-lang.org/getting-started/mix-otp/supervisor-and-application.html) methodology.

Otherwise, great use of pattern matching!